draft-gpew-priv-ppm.md

```diff
@@ -712,25 +719,22 @@ the first aggregate request. Shall we exclude it in subsequent requests
 somehow?] The remainder of the structure is dedicated to VDAF-specific request
 parameters.
 
-In order to provide replay protection, the leader is required to send aggregate
-sub-requests in ascending order, where the ordering on sub-requests is
-determined by the algorithm defined in {{anti-replay}}. Specifically, the leader
-constructs its request such that:
-
-* each sub-request follows the previous sub-request; and
-* the first sub-request follows the last sub-request in the previous aggregate
-  request.
+In order to provide replay protection, the leader preprocesses the set of
+reports carried by the AggregateReq as described in {{anti-replay}}. Any reports
+filtered out by this procedure MUST be ignored.
 
 The helper handles well-formed requests as follows. (As usual, malformed
 requests are handled as described in {{errors}}.) It first looks for PPM
-parameters corresponding to `AggregateReq.task_id`. It then filters out
-out-of-order sub-requests by ignoring any sub-request that does not follow the
-previous one (See {{anti-replay}}.)
+parameters corresponding to `AggregateReq.task_id`. It then preprocesses the
+sub-requests as described in {{anti-replay}}. Any sub-requests filtered out by
+this procedure MUST be ignored.
+
+In addition, the helper MUST ignore any report whose timestamp falls in a batch
+interval that has been aggregated at least once.
```

Comment:
You should include all the anti-replay text in the one place.  I read the previous piece and guessed that only the leader looks at timestamps.  But that isn't it.

Note also s/timestamp/nonce or "nonce contains a timestamp that"

Reply:
> You should include all the anti-replay text in the one place. I read the previous piece and guessed that only the leader looks at timestamps. But that isn't it.

Agreed. I tried but couldn't figure out how to do so without a more major refactor. One idea I had was to change the {{anti-replay}} section to speak more generally about "batch faults", i.e., instances where a report was included or excluded from a batch in a way that leaked more information about the measurement than I intended.

What I did instead was note above that the helper enforces the same thing.

Note that this text gets de-duplicated in https://github.com/abetterinternet/ppm-specification/pull/174.

> Note also s/timestamp/nonce or "nonce contains a timestamp that"

Done.